stress/shell/src/main/java/alluxio/stress/cli/client/StressClientIOBench.java

```diff
@@ -162,11 +182,132 @@ public ClientIOTaskResult runLocal() throws Exception {
+            || (type.equals("HDFSPacketReceiver") && methodName.equals("doRead"))) {
+          if (!timeToFirstByte.containsKey(methodName)) {
+            timeToFirstByte.put(methodName, new PartialResultStatistic());
+          }
+
+          final PartialResultStatistic statistic = timeToFirstByte.get(methodName);
+          statistic.mTimeToFirstByteNs.add(duration);
+          statistic.mNumSuccess += 1;
+
+          if (duration > statistic.mMaxTimeToFirstByteNs[0]) {
+            statistic.mMaxTimeToFirstByteNs[0] = duration;
+            Arrays.sort(statistic.mMaxTimeToFirstByteNs);
+          }
+        }
+      }
+    }
+    if (timeToFirstByte.containsKey("readChunk")) {
+      Collections.sort(timeToFirstByte.get("readChunk").mTimeToFirstByteNs);
+      SummaryStatistics stats = new SummaryStatistics(
+          timeToFirstByte.get("readChunk").mNumSuccess,
+          computeTimePercentileMS(timeToFirstByte.get("readChunk").mTimeToFirstByteNs),
+          computeTime99PercentileMS(timeToFirstByte.get("readChunk").mTimeToFirstByteNs),
+          computeMaxTimeMS(timeToFirstByte.get("readChunk").mMaxTimeToFirstByteNs));
+      return stats;
+    } else if (timeToFirstByte.containsKey("doRead")) {
+      Collections.sort(timeToFirstByte.get("doRead").mTimeToFirstByteNs);
+      SummaryStatistics stats = new SummaryStatistics(
+          timeToFirstByte.get("doRead").mNumSuccess,
+          computeTimePercentileMS(timeToFirstByte.get("doRead").mTimeToFirstByteNs),
+          computeTime99PercentileMS(timeToFirstByte.get("doRead").mTimeToFirstByteNs),
+          computeMaxTimeMS(timeToFirstByte.get("doRead").mMaxTimeToFirstByteNs));
+      return stats;
+    }
+    return new SummaryStatistics();
+  }
+
+  private float[] computeTimePercentileMS(ArrayList<Integer> rawTime) {
+    float[] timePercentileMS = new float[101];
+    int step = rawTime.size() / 100;
+
+    for (int index = 0; index < 101; index++) {
+      timePercentileMS[index] = (float) rawTime.get(step * index).intValue() / Constants.MS_NANO;
+    }
+
+    return timePercentileMS;
+  }
+
+  private float[] computeTime99PercentileMS(ArrayList<Integer> rawTime) {
+    int length = rawTime.size() - 1;
+    float[] timePercentileMS = new float[6];
+
+    for (int index = 0; index < 6; index++) {
+      timePercentileMS[index] = (float) rawTime.get(
+          (int) (length * (100.0 - 1.0 / Math.pow(10.0, index)) / 100.0))
+          / Constants.MS_NANO;
+    }
+
+    return timePercentileMS;
+  }
+
+  private float[] computeMaxTimeMS(long[] rawTime) {
+    int step = rawTime.length;
+    float[] timePercentileMS = new float[step];
+
+    for (int index = 0; index < step; index++) {
+      timePercentileMS[index] = (float) rawTime[index] / Constants.MS_NANO;
+    }
+
+    return timePercentileMS;
+  }
+
+  /**
+   * Result statistics of time to first byte measurement.
+   */
+  private final class PartialResultStatistic {
```

Comment:
Why isn't this class just using a `Histogram`? That is a useful library to handle all the times.

Reply:
Changed to `Histogram`.